### PR TITLE
[#3531] Show edit button if the report type is SAR

### PR DESCRIPTION
**Pull request by jessabean**
### Description
Addresses a bug reported during QA.

If a work plan has a closed out initiative, the SAR for that work plan should still allow the initiative to be editable, so users can report on the initiative.

### Related ticket(s)

CMDCT-3531

---
#### The easy way (if you trust me)
1. Go to the [cloudfront link](https://dlju3hjctuxjv.cloudfront.net/)
2. Log in as `stateuser@test.com`
3. View the SAR.
4. Go to the State or Territory-Specific Initiatives section. The initiative that was closed should still be marked as such, but the button next to it should say "Edit."
5. Click the Edit button, and then the Edit button for any of the entity rows. The form in the modal should also be editable.

#### The hard, manual way
1. Create a work plan.
6. Fill out the work plan, submit, and approve it.
7. Copy over the previous work plan.
8. In the copyover report, go to the State or Territory-Specific Initiatives section, and complete reporting for a particular initiative. At that point, the Edit button for the close-out information row will be enabled.
9. Close out the initiative.
10. Complete the work plan, submit and approve it.
11. Create a SAR for the work plan with the closed initiative.
12. Go to the State or Territory-Specific Initiatives section. The initiative that was closed should still be marked as such, but the button next to it should say "Edit."
13. Click the Edit button, and then the Edit buttons for any of the entity rows. The form in the modal should also be editable.

---
### Pre-review checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by gmrabian — COMMENTED** [review]
The first section with the modal is editable, but the second sections for progress and expenditures are not. The ticket indicates all sections should have full editing capabilities

Screenshot 2024-04-15 at 11 46 56 AM
Screenshot 2024-04-15 at 11 47 05 AM
Screenshot 2024-04-15 at 11 47 17 AM

File: services/ui-src/src/utils/testing/mockForm.tsx [hunk 1]
```diff
@@ -434,6 +434,7 @@ export const mockModalOverlayReportPageVerbiage = {
   enterReportText: "Mock enter report text",
   enterEntityDetailsButtonText: "Mock enter entity details button text",
   addEditModalEditTitle: "Mock AddEdit Modal Text",
+  reportProgressButtonText: "Mock report preogress button text",
```
Comment (gmrabian) [inline services/ui-src/src/utils/testing/mockForm.tsx:437]: nit: progress typo

File: services/ui-src/src/components/tables/getEntityStatus.test.ts [hunk 2]
```diff
@@ -231,6 +263,41 @@ describe("Entity status utilities", () => {
...
+            },
+          ],
+        },
+      } as ReportShape;
+      const isPdf = false;
+      const ignore: string[] = [];
+
+      const result = getInitiativeStatus(report, entity, isPdf, ignore);
+
+      expect(result).not.toBe(EntityStatuses.CLOSE);
```
Comment (gmrabian) [inline services/ui-src/src/components/tables/getEntityStatus.test.ts:298]: do we want to check for the expected status here? not closed is good though if not

File: services/ui-src/src/components/overlays/EntityDetailsOverlay.tsx [hunk 3]
```diff
@@ -55,6 +56,11 @@ export const EntityDetailsOverlay = ({
   const { full_name, state } = useStore().user ?? {};
   const { updateReport } = useContext(ReportContext);
   const [spinner, setSpinner] = useState<Boolean>();
+  const isSAR = report?.reportType === ReportType.SAR;
```
Comment (gmrabian) [inline services/ui-src/src/components/overlays/EntityDetailsOverlay.tsx:59]: this almost seems worth putting in context with how much we use it, but not necessary

**Review by gmrabian — DISMISSED** [review]
functionality on deployed env is good!
code here looks good.

File: services/ui-src/src/components/tables/getEntityStatus.test.ts (hunk 2 above)
Comment (jessabean) [reply services/ui-src/src/components/tables/getEntityStatus.test.ts:298]: We _could_, but that status kind of depends on whether the reporting info is complete or not ([list of statuses here](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/components/tables/EntityStatusIcon.tsx#L10-L15)), and I thought the more important thing to test is that we should _not_ be seeing the closed icon in the SAR. That icon, according to design, should only appear in the WP.

File: services/ui-src/src/components/overlays/EntityDetailsOverlay.tsx (hunk 3 above)
Comment (jessabean) [reply services/ui-src/src/components/overlays/EntityDetailsOverlay.tsx:59]: I'ma make a ticket to refactor that into context. 👍🏻

**Review by benmartin-coforma — DISMISSED** [review]
This code looks great! Nice work on that last test 👍

**Comment by gmrabian** [comment]
The test failing PR checks is passing locally 🤔 
I've tried running individually, as a file, and the entire suite multiple times

**Review by gmrabian — APPROVED** [review]

**Review by benmartin-coforma — APPROVED** [review]
I don't see any issues with the code here - and although I don't fully understand the spec, I understand this PR has had a lot of scrutiny from design, so I'm sure they caught everything they needed to.

**Pull request merged**